### PR TITLE
[Phase 1][Remove Stream/Optional usage] LPS-173075 remove_stream_and_optional_in_segments-service

### PR DESCRIPTION
**Pull request by jeffwu0724**
Hi Dante,
Resent form https://github.com/liferay-core-infra/liferay-portal/pull/3446
Thanks for checking!

### CODE REVIEW

File: modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java [hunk 1]
```diff
@@ -373,29 +373,42 @@ private Predicate<T> _getGTPredicate(
...
 		EntityField entityField, Object fieldValue) {
 
-		if (Objects.equals(entityField.getType(), EntityField.Type.STRING)) {
-			return p -> Stream.of(
-				(String[])p.get(_lambdaCollectionEntityField.getName())
-			).anyMatch(
-				c -> StringUtils.containsIgnoreCase(
-					String.valueOf(c), String.valueOf(fieldValue))
-			);
+		if (!Objects.equals(entityField.getType(), EntityField.Type.STRING)) {
```
Comment (jeffwu0724) [inline modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java:393]: flip this if-condition as mentioned

File: modules/apps/segments/segments-service/src/main/java/com/liferay/segments/service/impl/SegmentsExperimentServiceImpl.java [hunk 2]
```diff
@@ -220,20 +218,19 @@ public SegmentsExperiment runSegmentsExperiment(
 
 		_checkPermissions(segmentsExperiment, ActionKeys.UPDATE);
 
-		Set<Map.Entry<String, Double>> segmentsExperienceKeySplits =
-			segmentsExperienceKeySplitMap.entrySet();
+		Map<Long, Double> segmentsExperienceIdSplitMap = new HashMap<>();
 
-		Stream<Map.Entry<String, Double>> segmentsExperienceKeySplitsStream =
-			segmentsExperienceKeySplits.stream();
+		for (Map.Entry<String, Double> segmentsExperienceKeySplit :
```
Comment (jeffwu0724) [inline modules/apps/segments/segments-service/src/main/java/com/liferay/segments/service/impl/SegmentsExperimentServiceImpl.java:223]: inline segmentsExperienceKeySplitMap.entrySet().
But for this one, if we inline this in the for loop, don't we also have a performance loss if we need to iterate the for loop many times?

**Comment by jeffwu0724** [comment]
ci:test:tango

**Comment by Kvale1733** [comment]
@jeffwu0724 No regressions found.

File: modules/apps/segments/segments-service/src/main/java/com/liferay/segments/service/impl/SegmentsExperimentServiceImpl.java (hunk 2 above)
Comment (dantewang) [reply modules/apps/segments/segments-service/src/main/java/com/liferay/segments/service/impl/SegmentsExperimentServiceImpl.java:223]: @jeffwu0724 What do you mean by the loss? There's no difference between iterating on a variable and iterating directly on `map.entrySet()`.

**Comment by dantewang** [comment]
See https://github.com/liferay-core-infra/liferay-portal/pull/3639

File: modules/apps/segments/segments-service/src/main/java/com/liferay/segments/service/impl/SegmentsExperimentServiceImpl.java (hunk 2 above)
Comment (jeffwu0724) [reply modules/apps/segments/segments-service/src/main/java/com/liferay/segments/service/impl/SegmentsExperimentServiceImpl.java:223]: @dantewang 
My confusion is, if we iterate directly on `map.entrySet()`, don't we call this method `map.entrySet()` several times in the loop? While if we store it in a variable, we only call `map.entrySet()` once while declaring.
Thanks for your answer in advance!

Comment (dantewang) [reply modules/apps/segments/segments-service/src/main/java/com/liferay/segments/service/impl/SegmentsExperimentServiceImpl.java:223]: @jeffwu0724 It doesn't. The for-each loop is internally equivalent with

```
Iterator iterator = map.entrySet().iterator();

while (iterator.hasNext()) {
    Map.Entry entry = iterator.next();
    ......
}
```

One of my earliest tasks on Liferay Portal was to convert the above while-loop with foreach loop wherever possible :P

Comment (jeffwu0724) [reply modules/apps/segments/segments-service/src/main/java/com/liferay/segments/service/impl/SegmentsExperimentServiceImpl.java:223]: Thanks @dantewang for your detailed explanation!

**Pull request closed without merging**